### PR TITLE
Add docs comparing EKS Anywhere to EKS on Outposts

### PR DESCRIPTION
**Pull request by chrisnegus**
*Issue #, if available:* [Issue #1479](https://github.com/aws/eks-anywhere/issues/1479)

*Description of changes:* Update the [Compare EKS Anywhere and EKS](https://anywhere.eks.amazonaws.com/docs/concepts/eksafeatures/) page in the EKS Anywhere docs to describe differences between EKS Anywhere and Outposts.

### CODE REVIEW

File: docs/content/en/docs/concepts/eksafeatures.md [hunk 1]
```diff
@@ -51,3 +51,16 @@ To learn more about Amazon EKS, see [Amazon Elastic Kubernetes Service](https://
 | Control plane pricing                       | Free to download, paid support subscription option  | Hourly pricing per cluster |
 | AWS Support                       | Additional annual subscription (per cluster) for AWS support | Basic support included. Included in paid AWS support plans (developer, business, and enterprise)  |
 ||||
+
+### Comparing Amazon EKS Anywhere to Amazon EKS on Outposts
+
+Like EKS Anywhere, Amazon EKS on Outposts provides a means of running Kubernetes clusters using EKS software and customer on-premises hardware.
+The main difference is that Amazon provides the hardware with Outposts, while most EKS Anywhere providers leverage the customer's own hardware.
```
Comment (elylee) [inline docs/content/en/docs/concepts/eksafeatures.md:58]: To me, there are two main differences: the one you mentioned, and the second one is - EKS on Outpost provides an AWS-managed Kubernetes control plane versus with EKS-A, customers have to self-manage the Kubernetes control plane

Comment (csplinter) [reply docs/content/en/docs/concepts/eksafeatures.md]: +1 to what Lichun says. 

With Amazon EKS on Outposts, the Kubernetes control plane is fully managed by AWS. With EKS Anywhere, customers are responsible for managing the lifecycle of the Kubernetes control plane with EKS Anywhere automation tooling.

**Comment by csplinter** [comment]
I would also make a comment about the interface differences.

Customers can use Amazon EKS on Outposts with the same console, APIs, and tools they use to run Amazon EKS clusters in AWS Cloud. With EKS Anywhere, customers can use the eksctl CLI to manage their clusters, optionally connect their clusters to the EKS console for observability, and optionally use infrastructure as code tools such as Terraform and GitOps to manage their clusters. However, the primary interface for EKS Anywhere are the EKS Anywhere Custom Resources. Amazon EKS does not have a CRD-based interface today.

File: docs/content/en/docs/concepts/eksafeatures.md [hunk 2]
```diff
@@ -51,3 +51,19 @@ To learn more about Amazon EKS, see [Amazon Elastic Kubernetes Service](https://
 | Control plane pricing                       | Free to download, paid support subscription option  | Hourly pricing per cluster |
 | AWS Support                       | Additional annual subscription (per cluster) for AWS support | Basic support included. Included in paid AWS support plans (developer, business, and enterprise)  |
 ||||
+
+### Comparing Amazon EKS Anywhere to Amazon EKS on Outposts
+
+Like EKS Anywhere, Amazon EKS on Outposts provides a means of running Kubernetes clusters using EKS software and customer on-premises hardware.
+The main difference are that:
```
Comment (csplinter) [inline docs/content/en/docs/concepts/eksafeatures.md:58]: The main differences are:

File: docs/content/en/docs/concepts/eksafeatures.md [hunk 3]
```diff
@@ -51,3 +51,19 @@ To learn more about Amazon EKS, see [Amazon Elastic Kubernetes Service](https://
...
+
+* Amazon provides the hardware with Outposts, while most EKS Anywhere providers leverage the customer's own hardware.
+* Outposts have the option of having the EKS control plane run and be managed on the AWS cloud.
+
```
Comment (csplinter) [inline docs/content/en/docs/concepts/eksafeatures.md:62]: Amazon EKS on Outposts is a regional AWS service that requires a consistent, reliable connection from the Outpost to the AWS Region. EKS Anywhere is a standalone software offering that can run entirely disconnected from AWS Cloud, including air-gapped environments.

File: docs/content/en/docs/concepts/eksafeatures.md (hunk 2 above)
Comment (chrisnegus) [reply docs/content/en/docs/concepts/eksafeatures.md:58]: Done.

File: docs/content/en/docs/concepts/eksafeatures.md (hunk 3 above)
Comment (chrisnegus) [reply docs/content/en/docs/concepts/eksafeatures.md:62]: Done.

**Review by chrisnegus — COMMENTED** [review]
@csplinter Let me know if anything else is needed.

File: docs/content/en/docs/concepts/eksafeatures.md (hunk 1 above)
Comment (chrisnegus) [reply docs/content/en/docs/concepts/eksafeatures.md]: Done.

**Comment by csplinter** [comment]
@chrisnegus one minor comment, other than that looks good

File: docs/content/en/docs/concepts/eksafeatures.md [hunk 4]
```diff
@@ -51,3 +51,23 @@ To learn more about Amazon EKS, see [Amazon Elastic Kubernetes Service](https://
 | Control plane pricing                       | Free to download, paid support subscription option  | Hourly pricing per cluster |
 | AWS Support                       | Additional annual subscription (per cluster) for AWS support | Basic support included. Included in paid AWS support plans (developer, business, and enterprise)  |
 ||||
+
+### Comparing Amazon EKS Anywhere to Amazon EKS on Outposts
+
+Like EKS Anywhere, Amazon EKS on Outposts provides a means of running Kubernetes clusters using EKS software and customer on-premises hardware.
```
[the remark this answers is not in the document]
Comment (chrisnegus) [reply docs/content/en/docs/concepts/eksafeatures.md]: Done.

**Pull request merged**